app/src/main/java/com/codepath/chefster/activities/DishDetailsActivity.java

```diff
@@ -0,0 +1,100 @@
+package com.codepath.chefster.activities;
+
+import android.support.design.widget.TabLayout;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.os.Bundle;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.codepath.chefster.R;
+import com.codepath.chefster.fragments.IngredientsFragment;
+import com.codepath.chefster.fragments.ReviewFragment;
+import com.codepath.chefster.models.Dish;
+
+import org.parceler.Parcels;
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class DishDetailsActivity extends BaseActivity {
+    @BindView(R.id.vpDish)
+    ViewPager viewPagerDish;
+
+    @BindView(R.id.tlDish)
+    TabLayout tabLayoutDish;
+
+    @BindView(R.id.ivDishDetails)
+    ImageView ivDishDetails;
+
+    final static private int FIRST = 0;
+    final static private int SECOND = 1;
+    private Dish dish;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_dish_details);
+
+        ButterKnife.bind(this);
+
+        dish = Parcels.unwrap(getIntent().getParcelableExtra("selected_dish"));
+
+        Glide.with(this).load(dish.getThumbnails().get(0)).into(ivDishDetails);
+
+        setViewPager();
+    }
+
+    private void setViewPager() {
+        DishDetailsPagerAdapter adapter = new DishDetailsPagerAdapter(getSupportFragmentManager());
+        viewPagerDish.setAdapter(adapter);
+
+        tabLayoutDish.setupWithViewPager(viewPagerDish);
+    }
+
+    // This Class is for Fragments Adapter.
+    public class DishDetailsPagerAdapter extends FragmentPagerAdapter {
+        private String tabTitles[] = {"Ingredients", "Reviews"};
+
+        public DishDetailsPagerAdapter(FragmentManager fm) {
+            super(fm);
+        }
+
+        @Override
+        public Object instantiateItem(ViewGroup container, int position) {
+            return super.instantiateItem(container, position);
+        }
+
+        @Override
+        public Fragment getItem(int position) {
+            Bundle bundle = new Bundle();
+            if (position == FIRST) {
+                IngredientsFragment fragment = new IngredientsFragment();
```

Comment:
So this is a "teaching" comment. to be honest, I'm not sure, but pretty sure that you don't need to do `IngredientsFragment fragment = new IngredientsFragment();` every time you click on a tab. Please check if it's being called every time you click on a tab, because if it does, there could be a memory leak if you keep creating the same fragment, instead of recycling it from the viewpager. So from what I know, ViewPager has the instance of the fragment already so you need to check for if not null, use it, otherwise, create a new one. But I might be wrong. It's good for you to check in my opinion, we will all learn from it :)